### DISCUSSION TITLE
How does multi-line cell content work?

### DISCUSSION THREAD

**Discussion by chwt** [saulpw/visidata#2632] [Q&A] [answered]
Appreciating if someone could provide some examples / screenshots how that would look like? Should it also display multi-line text when in `v` mode?

(Currently when adding multi-line input via external editor, I don't see it displayed as such. Further, on reopen wrong new rows are shown instead.)

**Comment by midichef** [comment, marked as answer]
I believe multi-line display is for showing a cell that has a single long line of content, by wrapping it across multiple lines. It's not for editing a cell to make it take up more than 1 line. For example, if your goal is to put `"a\nb\nc\n"` in a single cell and have it consistently be displayed as 3 lines, I do not think visidata has that ability.

You say you reopen the file and see wrong new rows. What is the filetype?

I can imagine such a problem happening in a filetype like `tsv`, because as saulpw has said, [quoted newlines aren't allowed in 'true' tsv files](https://github.com/saulpw/visidata/issues/459#issuecomment-583811188). For tsv, if you add newlines in a cell and then save it, that will create new rows at the added newlines. But for `csv` files, you should be able to add newlines that keep their formatting upon saving and reloading.

**Reply by asteroidb612** [reply]
I also tried using the external editor to put newlines into a table, and I see dots where newlines would be: 
<img width="217" alt="Screenshot 2024-12-26 at 3 39 41 PM" src="https://github.com/user-attachments/assets/17d32407-6fc0-4c09-95d8-0bc57a8493a0" />

This looks the same after writing to a csv file and loading that file back in.

**Reply by saulpw** [reply]
This is intentional, to be able to show as much content as possible on one line (using `options.disp_oddspace`).  Use `v` to toggle multiline display which will be closer to what is expected.  However, even then, it's capped at `options.disp_max_wrap_lines` and the last line will again show as much content as possible.
